render/render.go

```diff
@@ -0,0 +1,39 @@
+package render
+
+import (
+	"bytes"
+	_ "embed"
+	"encoding/json"
+	"os"
+	"text/template"
+
+	"github.com/aquasecurity/trivy/pkg/types"
+	"golang.org/x/xerrors"
+)
+
+//go:embed template/html.tpl
+var htmlTmpl []byte
+
+func Render(fileName string, inputData []byte) error {
+	var output types.Report
+	if err := json.NewDecoder(bytes.NewReader(inputData)).Decode(&output); err != nil {
```

Comment:
The data already exists in memory, rather than coming from a stream, so you can use `json.Unmarshal`